CMakeLists.txt

```diff
@@ -276,11 +276,12 @@ if (WIN32)
         target_compile_options(
             mrdocs-core
             PUBLIC
-            /permissive-    # strict C++
-            /W4             # enable all warnings
-            /MP             # multi-processor compilation
-            /EHs            # C++ Exception handling
-            $<$<CONFIG:Debug>:/Oy-> # Disable frame pointer omission
+            /permissive-                    # strict C++
+            /W4                             # enable all warnings
+            $<$<CXX_COMPILER_ID:MSVC>:/MP>  # multi-processor compilation
+            /EHs                            # C++ Exception handling
+            $<$<CONFIG:Debug>:/Oy->         # Disable frame pointer omission
+            $<$<CXX_COMPILER_ID:Clang>:-Wno-unused-parameter>  # ClangCL warning enabled with /W4
```

Comment:
Shouldn't we fix the code triggering the warning instead of disabling the warning?

Reply:
I made a separate commit now. I initially added this because the warnings weren't present in the output of GCC and Clang on macOS/Linux.

Reply:
Yes. At some point we have to enable all these warnings there too. It's just that the project started without the these warnings and we always have higher priority stuff to work on.